src/runtime_src/core/tools/xbmgmt2/flash/xmc.h

```diff
@@ -16,117 +16,150 @@
-#define XMC_HOST_MSG_MSP432_MODE_ERR        0x03
-#define XMC_HOST_MSG_MSP432_FW_LENGTH_ERR   0x04
-#define XMC_HOST_MSG_BRD_INFO_MISSING_ERR   0x05
-
-#define XMC_MODE()      (readReg(XMC_REG_OFF_STATUS) & 0x3)
-#define BMC_MODE()      (readReg(XMC_REG_OFF_STATUS) >> 28)
-
-enum bmc_state {
-        BMC_STATE_UNKNOWN = 0,
-        BMC_STATE_READY,
-        BMC_STATE_BSL_UNSYNC,
-        BMC_STATE_BSL_SYNC
+enum class xmc_reg_offset
+{
+  magic         = 0x0,
+  version       = 0x4,
+  status        = 0x8,
+  error         = 0xc,
+  feature       = 0x10,
+  control       = 0x18,
+  packet_offset = 0x300,
+  packet_status = 0x304
 };
 
-#define XMC_READY       (0x1 << 0)
-#define XMC_STOPPED     (0x1 << 1)
-#define XMC_PAUSED      (0x1 << 2)
-
-enum xmc_packet_op {
-    XPO_UNKNOWN = 0,
-    XPO_MSP432_SEC_START,
-    XPO_MSP432_SEC_DATA,
-    XPO_MSP432_IMAGE_END,
-    XPO_BOARD_INFO,
-    XPO_MSP432_ERASE_FW
+enum class xmc_mask
+{
+  ctrl_error_clear = 1 << 1,
+  pkt_support      = 1 << 3,
+  pkt_owner        = 1 << 5,
+  pkt_error        = 1 << 26
+};
+
+enum class xmc_status
+{
+  ready    = 0x1 << 0,
+  stoppped = 0x1 << 1,
+  paused   = 0x1 << 2
 };
 
-const size_t xmcPktSize = (1024 / sizeof (uint32_t)) * 4; // In uint32_t
-struct xmcPkt {
-    // Make sure hdr is uint32_t aligned
-    struct xmcPktHdr {
-        unsigned payloadSize : 12;
-        unsigned reserved : 12;
-        unsigned opCode : 8;
-    } hdr;
-    uint32_t data[xmcPktSize - sizeof (hdr) / sizeof (uint32_t)];
+enum class xmc_host_error_msg 
+{
+  success          = 0x00,
+  bad_opcode       = 0x01,
+  unknown          = 0x02,
+  msp432_mode      = 0x03,
+  msp432_fw_length = 0x04,
+  brd_info_missing = 0x05
 };
-const size_t xmcMaxPayload =
-    xmcPktSize -sizeof (xmcPkt::xmcPktHdr) / sizeof (uint32_t); // In uint32_t
+
+enum class bmc_state
+{
+  unknown = 0,
+  ready,
+  bsl_unsync,
+  bsl_sync,
+  bsl_sync_not_upgradable,
+  ready_not_upgradable
+};
+
+enum class xmc_packet_op
+{
+  unknown = 0,
+  msp432_sec_start,
+  msp432_sec_data,
+  msp432_image_end,
+  board_info,
+  msp432_erase_fw
+};
+
+static const size_t xmc_pkt_size = (1024 / sizeof(uint32_t)) * 4;
+struct xmc_pkt
+{
+  // Make sure hdr is uint32_t aligned
+  struct xmc_pkt_header
+  {
+    unsigned payloadSize : 12;
+    unsigned reserved : 12;
+    unsigned opCode : 8;
+  } hdr;
+  uint32_t data[xmc_pkt_size - sizeof(hdr) / sizeof(uint32_t)];
+};
+static const size_t xmc_max_payload = xmc_pkt_size - sizeof(xmc_pkt::xmc_pkt_header) / sizeof(uint32_t);
+
+// ------------- C L A S S :  X M C _ F l a s h e r -----------------
 
 class XMC_Flasher
 {
-    struct ELARecord
-    {
-        unsigned mStartAddress;
-        unsigned mEndAddress;
-        unsigned mDataCount;
-        std::streampos mDataPos;
-        ELARecord() : mStartAddress(0), mEndAddress(0), mDataCount(0), mDataPos(0) {}
-    };
-
-    typedef std::list<ELARecord> ELARecordList;
-    ELARecordList mRecordList;
+  struct ELARecord
```

Comment:
Hmm...it looks like you just "beautified" this code....it does need to be cleaned up, but not in this way :^)